src/main/java/com/google/cloud/tools/internal/sdk/CloudSdkCommandFactory.java

```diff
@@ -0,0 +1,41 @@
+package com.google.cloud.tools.internal.sdk;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/**
+ * Created by appu on 3/29/16.
+ */
+public class CloudSdkCommandFactory {
+
+  public CloudSdk cloudSdk;
+
+  public CloudSdkCommandFactory(CloudSdk cloudSdk) {
+    this.cloudSdk = cloudSdk;
+  }
+
+  public String[] getGCloudAppCommand(List<String> args) {
```

Comment:
What are the advantages of having two get*Command instead of only one? They share a lot of code.